### ISSUE TITLE
Can't get bottom half arrow style to work when a macro is used.

### ISSUE THREAD

**Issue by kevintelus** [plantuml/plantuml#1867]
**Describe the bug**
Keep getting syntax error when trying to use the `-//` arrow style along with defined rectangles.

**To Reproduce**
This works fine:

```
@startuml

rectangle A1 as "component A1"
rectangle A2 as "component A2"

A1 -\\ A2 : "relationship"

@enduml
```

but this does not:

```
@startuml

rectangle A1 as "component A1"
rectangle A2 as "component A2"

A1 -// A2 : "relationship"

@enduml
```
**Expected behavior**
I expect to see two rectangles connected with a bottom-half arrow style.

**Screenshots**
![error](https://github.com/user-attachments/assets/9f37a38a-ae34-4f87-b679-1d512997fb7c)

**Desktop (please complete the following information):**

- OS: Windows
- PlantUML jar version - 1.2024.6

**Comment by jimnelson372** [comment]
@kevintelus,

Right.   There appears to be only one type of "half arrow" in these "Description Diagrams" (of which Component Diagrams are included), which is why it is currently showing a syntax arrow rather than a stack dump.

Here is the documentation for the types of arrows that apply.  Though this is the documentation for "Deployment Diagrams" it's the same set of arrow heads for Component Diagrams.

https://plantuml.com/deployment-diagram#e281b137af34842a

![image](https://github.com/user-attachments/assets/fa51c132-3ffd-4046-b880-b847cd9b5772)

Notice that the only half arrow is on f7 to b7.   Most of these arrow heads also work (when reversed) for arrows going the other diction, though in that case there are no half arrows currently implemented.

Though I've worked on the arrow heads for sequence diagrams, I'm not familiar with the history of PlanUML's implementation of Component Diagrams to know why only the one half arrow type is allowed -- if there was a reason to restrict it.   If the other side were allowed, it would also be reasonable to ask for half side filled arrows, etc., for -/ or -\, just as they exist on sequence diagrams.

I see the implementation for this half arrow, and it seems like adding the other versions might be straight forward to add, but I haven't yet tried working on it.

Jim N.
@jimnelson372

**Comment by kevintelus** [comment]
Hi @jimnelson372 - I appreciate the response.  Perhaps you're right, maybe there's a reason to restrict it.  The context for me was that I am trying to use the [Archimate Association Relationship](https://pubs.opengroup.org/architecture/archimate31-doc/chap05.html#_Toc10045321) notation which is denoted by a half arrow.  I noticed in the diagram example it shows both an upper half arrow and a bottom half arrow.  It appears that when the relationship is reversed, it seems like it's represented by a bottom half arrow.  So perhaps I may be able to use it as-is after all.

![image](https://github.com/user-attachments/assets/4543040c-69e8-4cb7-9790-725faaf8854d)

**Comment by The-Lum** [comment]
Hi all,

_[This is an Issue Review]_ 👀 

This is now fixed, on [v1.2024.8](https://github.com/plantuml/plantuml/releases/tag/v1.2024.8)
- Thanks @kevintelus for the report.
- Thanks PlantUML team and @arnaudroques for your work.

Regards.
